Ikuyadeu/vscode-R
# Enhance R markdown support

**What problem did you solve?**

Close #428, #49, #261 

* [x] Run chunk and Run above command
* [x] Highlight code chunks
* [x] Respect chunk options (Run above should skip chunks with `eval=FALSE`)
* [x] Command to run current chunk
* [x] Completions for chunk options

**(If you have)Screenshot**

![Kapture 2020-10-28 at 22 25 08](https://user-images.githubusercontent.com/4662568/97449414-99a7b780-196c-11eb-9d2e-2c8eb5804d54.gif)


How to test:

1. Create a Rmd document with a number of code chunks (using `r chunk` snippet)
2. Above each chunk there are two CodeLens items: Run Chunk and Run Above.
3. Each chunk has background highlighting.
4. Click Run Chunk and the chunk code is sent to terminal
5. Click Run Above and the code above the chunk is combined and sent to terminal
6. If a chunk has option `eval=FALSE` then Run Above will not include such chunks.
7. Put cursor in a chunk and run `R: Run Current Chunk` and the chunk is sent to terminal.8
8. If cursor is outside any chunk, run `R: Run Current Chunk` and nothing happens.
9. Pressing `Ctrl+shift+enter` should trigger `R: Run Current Chunk` by default (same as RStudio).
10. Editing chunk header will trigger chunk options completion and selecting an item will insert `{label}=` to the editor.
